nix/wire-server.nix

```diff
@@ -0,0 +1,271 @@
+pkgs:
+let lib = pkgs.lib;
+    hlib = pkgs.haskell.lib;
+    withCleanedPath = drv:
+      hlib.overrideCabal drv (old: {
+        src = lib.cleanSourceWith {
+          src = old.src;
+          filter = path: type:
+            let baseName = baseNameOf (toString path);
+            in baseName != "dist";
+        };
+      });
+
+    gitignoreSource =
+      let gitignoreSrc = pkgs.fetchFromGitHub {
```

Comment:
There should be a gitignore source vendored into nixpkgs, which we can probably use instead.

Reply:
See `doc/functions/nix-gitignore.section.md` in nixpkgs

Reply:
If we don't want to specify the path to the root .gitignore when calling `gitignoreSource`, we might also be able to use `nix-gitignore.gitignoreSourcePure` with a small hardcoded list of paths to ignore, that impure local haskell builds create in the workdir.

Reply:
`nix-gitignore.gitignoreSource` just fails if there are no `.gitignore` files in the package directory, the one we have right now traverses parent directories to figure out the gitignore. I don't want to duplicate the list by using `.gitignoreSourcePure` because then someone will forget to add things here and be surprised. So, I think its not so bad to rely on `hercules-ci/gitignore.nix`.

Reply:
Then we'd have to write code to figure out relative path of the repo_root from every package. Why is using the gitignore from hercules-ci so bad?

Reply:
Hmmh, I assumed we have like one expression for each wire-server component, and these files are committed into the repository, so the path is known and won't change. I wasn't aware we codegen here as well?

Reply:
The expression for each haskell component is generated, this changes everytime there is a change in the cabal file.